neuro_evolution_ctrnn/brains/i_layer_based_brain.py

```diff
@@ -0,0 +1,190 @@
+        input_size = self._size_from_space(input_space)
+        output_size = self._size_from_space(output_space)
+        hidden_structure: List[int] = config.hidden_structure
+
+        # initialize weights out of individual
+
+        individual_index = 0  # progress index
+        # initialize emphty
+        self.weight_ih = []  # Weights for weighted input values
+        self.weight_hh = []  # Weights for weighted stored values
+        self.bias_h = []  # Biases
+        self.hidden = []  # Initial values for state storage
+        self.layer_output = []  # Weights from output last Layer to output nodes
+        number_gates = self.get_number_gates()
+
+        # iterate for all given layers in the structure
+        for layer in range(len(hidden_structure)):
+
+            # Matrices for weighted input values in calculations
+
+            if layer == 0:  # The first Layer don't has an output from the previous layer, but the input values
+                self.weight_ih.append(
+                    np.array(
+                        individual[
+                            individual_index:
+                            individual_index + number_gates * hidden_structure[layer] * input_size
+                        ]
+                    ).reshape((number_gates, hidden_structure[layer], input_size))
+                )
+                individual_index += number_gates * input_size * hidden_structure[0]
+            else:
+                self.weight_ih.append(
+                    np.array(
+                        individual[
+                            individual_index:
+                            individual_index + number_gates * hidden_structure[layer] * hidden_structure[layer - 1]
+                        ]
+                    ).reshape((number_gates, hidden_structure[layer], hidden_structure[layer - 1]))
+                )
+                individual_index += number_gates * hidden_structure[layer - 1] * hidden_structure[layer]
+
+            # Matrices for weighted state values in calculations
+            if config.diagonal_hidden_to_hidden:  # Whether each neuron can only access its own state
+                self.weight_hh.append(
+                    [np.diag(individual[
+                             individual_index + k * hidden_structure[layer] * hidden_structure[layer]:
+                             individual_index + k * hidden_structure[layer] * hidden_structure[layer] +
+                             hidden_structure[layer]])
+                     for k in range(number_gates)
+                     ]
+                )
+                individual_index += number_gates * hidden_structure[layer]
+            else:
+                self.weight_hh.append(
+                    np.array(
+                        individual[
+                            individual_index:
+                            individual_index + number_gates * hidden_structure[layer] * hidden_structure[layer]
```

Comment:
ich habe so ein Gefühl, als wäre hier ein Fehler. Ich habe so ein gefühl, als würde der ganze code hier nur funktionieren, wenn alle hidden layers gleich groß sind. 

Hier z.B. sollte das eine `hidden_structure[layer]`  eher `hidden_structure[layer+1]` sein, oder? 

(gibt noch mehr stellen, wo ich ein bezug zur nächsten layer erwartet hätte, aber keiner da war)

Reply:
ne, das ist gut getestet. In dem Teil geht es nur um die Schicht-internen Gewichte

Reply:
ah, ok. Danke